File: Programming_Basics_with_Python/while_loop_exercise/02._exam_preparation.py
failed_limit = int(input())
last_command = ''
failed_number = 0
total_tasks = 0
total_grades = 0
has_failed = True

while failed_number < failed_limit:
    command = input()
    if command == "Enough":
        has_failed = False
        break
    grade = int(input())
    total_tasks += 1
    total_grades += grade
    last_command = command
    if grade <= 4:
        failed_number += 1


average = total_grades / total_tasks

if failed_number >= failed_limit:
    print(f"You need a break, {failed_number} poor grades.")
else:
    print(f"Average score: {average:.2f}")
    print(f"Number of problems: {total_tasks}")
    print(f"Last problem: {last_command}")

File: Programing_Basics_with_JS/whileLoopExercise/02.ExamPreparation.js
function examPreparation(input) {
    let index = 0;
    let numbersOfBadGrades = Number(input[index]);
    index++;

    let allProblemsCount = 0;
    let allGrades = 0;
    let badGrades = 0;
    let problemName = '';
    let needBreak = false;

    let command = input[index];
    index++;

    while(command !== 'Enough') {
        problemName = command;
        allProblemsCount++;
        let grade = Number(input[index]);
        index++;
        allGrades += grade;
        
        if(grade <= 4){
            badGrades++;
            if(badGrades === numbersOfBadGrades) {
                needBreak = true;
                break;
            }
        }

        command = input[index];
        index++;
    }

    let averageGrade = allGrades / allProblemsCount;

    if(needBreak) {
        console.log(`You need a break, ${badGrades} poor grades.`);
    } else {
        console.log(`Average score: ${averageGrade.toFixed(2)}`);
        console.log(`Number of problems: ${allProblemsCount}`);
        console.log(`Last problem: ${problemName}`);
    }
}

examPreparation(["3",
"Money",
"6",
"Story",
"4",
"Spring Time",
"5",
"Bus",
"6",
"Enough"])

examPreparation(["2",
"Income",
"3",
"Game Info",
"6",
"Best Player",
"4"])
